### PR TITLE
DOC: configure and add spelling wordlist; fix typos, apply Sphinx syntax

### PR DESCRIPTION
**Pull request by mwtoews**
Changes mostly affect documentation, although a few are within the source code too.

Feel free to comment on any changes here.

This does not include potential changes for #2670 and #2701. Issue #2704 is resolved and added to the word list as correct.

### CODE REVIEW

**Comment by rouault** [comment]
is the  docs/source/spelling_wordlist.txt  file used by the "make spelling" target in docs/ ?  And that doesn't fix all issues reported by it, right  (looking at https://github.com/OSGeo/PROJ/pull/2705/checks?check_run_id=2482452219 ) ?

**Comment by mwtoews** [comment]
The `make spelling` target is passed through a catch-all target in [docs/Makefile](https://github.com/OSGeo/PROJ/blob/0f9e6bf1a21a6548501870c7c090811b199a57ff/docs/Makefile#L41-L43), so the configuration reads docs/source/spelling_wordlist.txt each time it is called. The missing target of course is .doxygen_up_to_date, so any edits to Doxygen source code files needs to be manually triggered before, i.e. `make html`.

As for the remaining 1690 misspelled words in this PR (so far), it's an improvement over the current 3626 misspelled words in the main branch. I don't expect this to get close to zero. Also the results vary considerably, depending on the system and locale, as I only have 765 misspelled words with this PR. I might add a few more to spelling_wordlist.txt based on what I see from the CI output.

File: src/iso19111/operation/transformation.cpp [hunk 1]
```diff
@@ -366,7 +366,7 @@ TransformationNNPtr Transformation::create(
 
 // ---------------------------------------------------------------------------
 
-/** \brief Instantiate a transformation ands its OperationMethod.
+/** \brief Instantiate a transformation and it's OperationMethod.
```
Comment (rouault) [inline src/iso19111/operation/transformation.cpp]: that one should be reverted

Comment (rouault) [reply src/iso19111/operation/transformation.cpp]: oops no, I see that the actual fix was ands->and.  I meant that the it's should be its

**Comment by rouault** [comment]
> The make spelling target is passed through a catch-all target in docs/Makefile, so the configuration reads docs/source/spelling_wordlist.txt each time it is called. 

Thanks for clearing that up, and the many fixes!

File: src/iso19111/operation/transformation.cpp (hunk 1 above)
Comment (rouault) [inline src/iso19111/operation/transformation.cpp]:
```suggestion
/** \brief Instantiate a transformation and its OperationMethod.
```

**Review by rouault — APPROVED** [review]

File: src/iso19111/operation/transformation.cpp (hunk 1 above)
Comment (mwtoews) [reply src/iso19111/operation/transformation.cpp]: thanks, amended edit

**Pull request merged**